execution/executors/shim/process.go

```diff
@@ -31,7 +32,26 @@ type newProcessOpts struct {
 	execution.StartProcessOpts
 }
 
+func validateNewProcessOpts(o newProcessOpts) error {
+	if o.shimBinary == "" {
+		return fmt.Errorf("invalid shimBinary: %q", o.shimBinary)
```

Comment:
Also, since we're only checking for empty string, the error can simply be something like "shim binary not specified" or something equivalent (same for the other checks).